File: android-exercises/6 - recycler_view/RecyclerView_java/app/src/main/java/com/bugrahankaramollaoglu/recyclerview_java/ColorAdapter.java
package com.bugrahankaramollaoglu.recyclerview_java;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bugrahankaramollaoglu.recyclerview_java.databinding.RecyclerRowBinding;

public class ColorAdapter extends RecyclerView.Adapter<ColorAdapter.ColorHolder> {
    private final String[] colorsList;

    public ColorAdapter(String[] colorsList) {
        this.colorsList = colorsList;
    }

    @NonNull
    @Override
    public ColorHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RecyclerRowBinding recyclerRowBinding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.getContext()),
                parent, false);
        return new ColorHolder(recyclerRowBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ColorHolder holder, int position) {
        holder.binding.recyclerViewTextView.setText(colorsList[position]);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int selectedItem = holder.getAdapterPosition();
                String selectedItemName = colorsList[selectedItem];
                Toast.makeText(holder.itemView.getContext(), "you touched: " + selectedItemName, Toast.LENGTH_SHORT).show();
            }
        });
    }


    @Override
    public int getItemCount() {
        return colorsList.length;
    }

    public class ColorHolder extends RecyclerView.ViewHolder {

        private RecyclerRowBinding binding;

        public ColorHolder(RecyclerRowBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

}

File: android-exercises/6 - recycler_view/RecyclerView_kotlin/app/src/main/java/com/bugrahankaramollaoglu/recyclerview_kotlin/ColorAdapter.kt
package com.bugrahankaramollaoglu.recyclerview_kotlin

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.bugrahankaramollaoglu.recyclerview_kotlin.databinding.RecycleRowBinding

class ColorAdapter(val color_list: ArrayList<String>) :
    RecyclerView.Adapter<ColorAdapter.ColorHolder>() {

    class ColorHolder(val binding: RecycleRowBinding) : RecyclerView.ViewHolder(binding.root) {

    }

    // ColorHolder ilk oluşturulduğunda ne olacak
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ColorHolder {
        val binding = RecycleRowBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ColorHolder(binding)
    }

    // bağlanacak ColorHolder sayısı
    override fun getItemCount(): Int {
        return color_list.size
    }

    override fun onBindViewHolder(holder: ColorHolder, position: Int) {
        holder.binding.recycleViewTextView.text = color_list[position]

        holder.itemView.setOnClickListener {
            Toast.makeText(
                holder.itemView.context,
                "you chose ${color_list[position]}",
                Toast.LENGTH_SHORT
            ).show()
        }
    }
}